### ISSUE TITLE
Memory Leaks

### ISSUE THREAD

**Issue by PaulBGD** [cr0ssVtW/SRPartyStuff#1]
I noticed quite a few memory leaks throughout the code. All cooldown data and such should be removed from the HashMaps after the player has quit. This is probably the reason why restarts were so frequent.

**Comment by cr0ssVtW** [comment]
Absolutely. Looking over it again after its all been said and done, definitely a lot of sloppy code. I think at one point I went back and tried to fix up the obvious stuff, but was just overwhelmed with other things so I never got far. I wanted to completely re-do this entire Party/Class system but never got around to it!

**Comment by PaulBGD** [comment]
Thanks for responding. Yeah, it could definitely be improved by some sort of abstract system. Less EventHandlers would be great as well, they could easily be replaced by some onDamage or onDamageOther methods. It'd allow for less method calls. As for storing data for cooldowns, have you considered a player wrapper with a temporary data structure? I've been using that a lot lately, very efficient.
